// react-google-maps-api/lib/components/drawing/Marker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Marker = void 0;

var _react = _interopRequireWildcard(require("react"));

var _helper = require("../../utils/helper");

var _mapContext = _interopRequireDefault(require("../../map-context"));

var _proptypes = require("../../proptypes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var eventMap = {
  onAnimationChanged: 'animation_changed',
  onClick: 'click',
  onClickableChanged: 'clickable_changed',
  onCursorChanged: 'cursor_changed',
  onDblClick: 'dblclick',
  onDrag: 'drag',
  onDragEnd: 'dragend',
  onDraggableChanged: 'draggable_changed',
  onDragStart: 'dragstart',
  onFlatChanged: 'flat_changed',
  onIconChanged: 'icon_changed',
  onMouseDown: 'mousedown',
  onMouseOut: 'mouseout',
  onMouseOver: 'mouseover',
  onMouseUp: 'mouseup',
  onPositionChanged: 'position_changed',
  onRightClick: 'rightclick',
  onShapeChanged: 'shape_changed',
  onTitleChanged: 'title_changed',
  onVisibleChanged: 'visible_changed',
  onZindexChanged: 'zindex_changed'
};
var updaterMap = {
  animation: function animation(instance, _animation) {
    instance.setAnimation(_animation);
  },
  clickable: function clickable(instance, _clickable) {
    instance.setClickable(_clickable);
  },
  cursor: function cursor(instance, _cursor) {
    instance.setCursor(_cursor);
  },
  draggable: function draggable(instance, _draggable) {
    instance.setDraggable(_draggable);
  },
  icon: function icon(instance, _icon) {
    instance.setIcon(_icon);
  },
  label: function label(instance, _label) {
    instance.setLabel(_label);
  },
  map: function map(instance, _map) {
    instance.setMap(_map);
  },
  opacity: function opacity(instance, _opacity) {
    instance.setOpacity(_opacity);
  },
  options: function options(instance, _options) {
    instance.setOptions(_options);
  },
  position: function position(instance, _position) {
    instance.setPosition(_position);
  },
  shape: function shape(instance, _shape) {
    instance.setShape(_shape);
  },
  title: function title(instance, _title) {
    instance.setTitle(_title);
  },
  visible: function visible(instance, _visible) {
    instance.setVisible(_visible);
  },
  zIndex: function zIndex(instance, _zIndex) {
    instance.setZIndex(_zIndex);
  }
};

var Marker =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(Marker, _PureComponent);

  function Marker() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Marker);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Marker)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "registerEvents", []);

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      marker: null
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidMount", function () {
      var marker = new google.maps.Marker(Object.assign({
        map: _this.context
      }, _this.props.options));

      _this.setState(function () {
        return {
          marker: marker
        };
      }, function () {
        _this.state.marker.setMap(_this.context);

        _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
          updaterMap: updaterMap,
          eventMap: eventMap,
          prevProps: {},
          nextProps: _this.props,
          instance: _this.state.marker
        });
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidUpdate", function (prevProps) {
      (0, _helper.unregisterEvents)(_this.registeredEvents);
      _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
        updaterMap: updaterMap,
        eventMap: eventMap,
        prevProps: prevProps,
        nextProps: _this.props,
        instance: _this.state.marker
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentWillUnmount", function () {
      (0, _helper.unregisterEvents)(_this.registeredEvents);
      _this.state.marker && _this.state.marker.setMap(null);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "render", function () {
      return _react.default.createElement(_react.Fragment, null, _this.props.children);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getAnimation", function () {
      return _this.state.marker.getAnimation();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getClickable", function () {
      return _this.state.marker.getClickable();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getCursor", function () {
      return _this.state.marker.getCursor();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getDraggable", function () {
      return _this.state.marker.getDraggable();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getIcon", function () {
      return _this.state.marker.getIcon();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getLabel", function () {
      return _this.state.marker.getLabel();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getMap", function () {
      return _this.state.marker.getMap();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getOpacity", function () {
      return _this.state.marker.getOpacity();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getPosition", function () {
      return _this.state.marker.getPosition();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getShape", function () {
      return _this.state.marker.getShape();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getTitle", function () {
      return _this.state.marker.getTitle();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getVisible", function () {
      return _this.state.marker.getVisible();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getZIndex", function () {
      return _this.state.marker.getZIndex();
    });

    return _this;
  }

  return Marker;
}(_react.PureComponent);

exports.Marker = Marker;

_defineProperty(Marker, "contextType", _mapContext.default);

var _default = Marker;
exports.default = _default;
