// react-google-maps-api/lib/components/places/SearchBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SearchBox = void 0;

var _react = require("react");

var _reactDom = require("react-dom");

var _invariant = _interopRequireDefault(require("invariant"));

var _helper = require("../../utils/helper");

var _mapContext = _interopRequireDefault(require("../../map-context"));

var _proptypes = require("../../proptypes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var eventMap = {
  onPlacesChanged: 'places_changed'
};
var updaterMap = {
  bounds: function bounds(instance, _bounds) {
    instance.setBounds(_bounds);
  }
};

var SearchBox =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(SearchBox, _PureComponent);

  function SearchBox(props, context) {
    var _this;

    _classCallCheck(this, SearchBox);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(SearchBox).call(this, props, context));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "registeredEvents", []);

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      searchBox: null,
      mountControlIndex: 0
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidMount", function () {
      var searchBox = new google.maps.places.SearchBox(_this.props.containerElement.querySelector('input'), Object.assign({
        map: _this.context
      }, _this.props.options));

      _this.setState(function (prevState, prevProps) {
        return {
          searchBox: searchBox,
          mountControlIndex: -1 + _this.context.controls[prevProps.controlPosition].push(prevProps.containerElement.firstChild)
        };
      }, function () {
        _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
          updaterMap: updaterMap,
          eventMap: eventMap,
          prevProps: {},
          nextProps: _this.props,
          instance: _this.state.searchBox
        });
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidUpdate", function (prevProps) {
      (0, _helper.unregisterEvents)(_this.registeredEvents);

      if (_this.props.controlPosition !== prevProps.controlPosition) {
        if (typeof _this.props.controlPosition === 'number') {
          _this.setState(function (prevState, prevProps) {
            return {
              mountControlIndex: -1 + _this.context.controls[prevProps.controlPosition].push(prevProps.containerElement.firstChild)
            };
          });
        }
      }

      _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
        updaterMap: updaterMap,
        eventMap: eventMap,
        prevProps: prevProps,
        nextProps: _this.props,
        instance: _this.state.searchBox
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentWillUnmount", function () {
      (0, _helper.unregisterEvents)(_this.registeredEvents);

      if (typeof _this.props.controlPosition === 'number') {
        var child = _this.context.controls[_this.props.controlPosition].removeAt(_this.state.mountControlIndex);

        if (child !== undefined) {
          _this.props.containerElement.appendChild(child);
        }
      }
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "render", function () {
      return (0, _reactDom.createPortal)(_react.Children.only(_this.props.children), _this.state.containerElement);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getBounds", function () {
      return _this.state.searchBox.getBounds();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getPlaces", function () {
      return _this.state.searchBox.getPlaces();
    });

    (0, _invariant.default)(google.maps.places, 'Did you include "libraries=places" in the URL?');
    return _this;
  }

  return SearchBox;
}(_react.PureComponent);

exports.SearchBox = SearchBox;

_defineProperty(SearchBox, "contextType", _mapContext.default);

var _default = SearchBox;
exports.default = _default;
