// react-google-maps-api/lib/components/addons/InfoBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.InfoBox = void 0;

var _react = require("react");

var _reactDom = require("react-dom");

var _invariant = _interopRequireDefault(require("invariant"));

var _proptypes = require("../../proptypes");

var _helper = require("../../utils/helper");

var _mapContext = _interopRequireDefault(require("../../map-context"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var open = function open(infoBox, anchor) {
  if (anchor) {
    infoBox.open(infoBox.getMap(), anchor);
  } else if (infoBox.getPosition()) {
    infoBox.open(infoBox.getMap());
  } else {
    (0, _invariant.default)(false, 'You must provide either an anchor (typically render it inside a <Marker>) or a position props for <InfoBox>.');
  }
};

var eventMap = {
  onCloseClick: 'closeclick',
  onDomReady: 'domready',
  onContentChanged: 'content_changed',
  onPositionChanged: 'position_changed',
  onZindexChanged: 'zindex_changed'
};
var updaterMap = {
  options: function options(instance, _options) {
    instance.setOptions(_options);
  },
  position: function position(instance, _position) {
    instance.setPosition(_position);
  },
  visible: function visible(instance, _visible) {
    instance.setVisible(_visible);
  },
  zIndex: function zIndex(instance, _zIndex) {
    instance.setZIndex(_zIndex);
  }
};

var InfoBox =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(InfoBox, _PureComponent);

  function InfoBox() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, InfoBox);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(InfoBox)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "registeredEvents", []);

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      infoBox: null
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidMount", function () {
      var _require = require('google-maps-infobox'),
          GoogleMapsInfobox = _require.InfoBox;

      var infoBox = new GoogleMapsInfobox();

      _this.setState(function () {
        return {
          infoBox: infoBox
        };
      }, function () {
        _this.state.infoBox.setMap(_this.context);

        _this.contentElement = document.createElement('div');

        _this.state.infoBox.setContent(_this.contentElement);

        open(_this.state.infoBox, _this.props.anchor);
        _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
          updaterMap: updaterMap,
          eventMap: eventMap,
          prevProps: {},
          nextProps: _this.props,
          instance: _this.state.infoBox
        });
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidUpdate", function (prevProps) {
      (0, _helper.unregisterEvents)(_this.registeredEvents);
      _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
        updaterMap: updaterMap,
        eventMap: eventMap,
        prevProps: prevProps,
        nextProps: _this.props,
        instance: _this.state.infoBox
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentWillUnmount", function () {
      (0, _helper.unregisterEvents)(_this.registeredEvents);

      if (_this.state.infoBox) {
        _this.state.infoBox.setMap(null);
      }
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "render", function () {
      return (0, _reactDom.createPortal)(_react.Children.only(_this.props.children), _this.contentElement);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getPosition", function () {
      return _this.state.infoBox.getPosition();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getVisible", function () {
      return _this.state.infoBox.getVisible();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getZIndex", function () {
      return _this.state.infoBox.getZIndex();
    });

    return _this;
  }

  return InfoBox;
}(_react.PureComponent);

exports.InfoBox = InfoBox;

_defineProperty(InfoBox, "contextType", _mapContext.default);

var _default = InfoBox;
exports.default = _default;
