// react-google-maps-api/lib/components/addons/MarkerWithLabel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.MarkerWithLabel = void 0;

var _react = require("react");

var _markerwithlabel = _interopRequireDefault(require("markerwithlabel"));

var _reactDom = require("react-dom");

var _helper = require("../../utils/helper");

var _MarkerCluster = require("./MarkerCluster");

var _proptypes = require("../../proptypes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var eventMap = {
  onDblClick: 'dblclick',
  onDragEnd: 'dragend',
  onDragStart: 'dragstart',
  onMouseDown: 'mousedown',
  onMouseOut: 'mouseout',
  onMouseOver: 'mouseover',
  onMouseUp: 'mouseup',
  onRightClick: 'rightclick',
  onAnimationChanged: 'animation_changed',
  onClick: 'click',
  onClickableChanged: 'clickable_changed',
  onCursorChanged: 'cursor_changed',
  onDrag: 'drag',
  onDraggableChanged: 'draggable_changed',
  onFlatChanged: 'flat_changed',
  onIconChanged: 'icon_changed',
  onPositionChanged: 'position_changed',
  onShapeChanged: 'shape_changed',
  onTitleChanged: 'title_changed',
  onVisibleChanged: 'visible_changed',
  onZindexChanged: 'zindex_changed'
};
var updaterMap = {
  labelAnchor: function labelAnchor(instance, _labelAnchor) {
    instance.set("labelAnchor", _labelAnchor);
  },
  labelClass: function labelClass(instance, _labelClass) {
    instance.set("labelClass", _labelClass);
  },
  labelStyle: function labelStyle(instance, _labelStyle) {
    instance.set("labelStyle", _labelStyle);
  },
  labelVisible: function labelVisible(instance, _labelVisible) {
    instance.set("labelVisible", _labelVisible);
  },
  animation: function animation(instance, _animation) {
    instance.setAnimation(_animation);
  },
  clickable: function clickable(instance, _clickable) {
    instance.setClickable(_clickable);
  },
  cursor: function cursor(instance, _cursor) {
    instance.setCursor(_cursor);
  },
  draggable: function draggable(instance, _draggable) {
    instance.setDraggable(_draggable);
  },
  icon: function icon(instance, _icon) {
    instance.setIcon(_icon);
  },
  label: function label(instance, _label) {
    instance.setLabel(_label);
  },
  opacity: function opacity(instance, _opacity) {
    instance.setOpacity(_opacity);
  },
  options: function options(instance, _options) {
    instance.setOptions(_options);
  },
  place: function place(instance, _place) {
    instance.setPlace(_place);
  },
  position: function position(instance, _position) {
    instance.setPosition(_position);
  },
  shape: function shape(instance, _shape) {
    instance.setShape(_shape);
  },
  title: function title(instance, _title) {
    instance.setTitle(_title);
  },
  visible: function visible(instance, _visible) {
    instance.setVisible(_visible);
  },
  zIndex: function zIndex(instance, _zIndex) {
    instance.setZIndex(_zIndex);
  }
};

var MarkerWithLabel =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(MarkerWithLabel, _PureComponent);

  function MarkerWithLabel() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, MarkerWithLabel);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(MarkerWithLabel)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "registeredEvents", []);

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      markerWithLabel: null
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidMount", function () {
      var NativeMarkerWithLabel = (0, _markerwithlabel.default)(google.maps);
      var markerWithLabel = new NativeMarkerWithLabel();

      _this.setState(function () {
        return {
          markerWithLabel: markerWithLabel
        };
      }, function () {
        _this.state.markerWithLabel.setMap(_this.context.map);

        _this.context.cluster.addMarker(_this.state.markerWithLabel, _this.props.noRedraw);

        _this.containerElement = document.createElement('div');

        _this.state.markerWithLabel.set('labelContent', _this.containerElement);
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidUpdate", function (prevProps) {
      (0, _helper.unregisterEvents)(_this.registeredEvents);
      _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
        updaterMap: updaterMap,
        eventMap: eventMap,
        prevProps: prevProps,
        nextProps: _this.props,
        instance: _this.state.markerWithLabel
      });

      _this.state.markerWithLabel.repaint();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentWillUnmount", function () {
      (0, _helper.unregisterEvents)(_this.registeredEvents);

      if (_this.state.markerWithLabel) {
        if (_this.context.cluster) {
          _this.context.cluster.removeMarker(_this.state.markerWithLabel, _this.props.noRedraw);
        }

        _this.state.markerWithLabel.setMap(null);
      }
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "render", function () {
      return (0, _reactDom.createPortal)(_react.Children.only(_this.props.children), _this.containerElement);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getAnimation", function () {
      return _this.state.markerWithLabel.getAnimation();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getClickable", function () {
      return _this.state.markerWithLabel.getClickable();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getCursor", function () {
      return _this.state.markerWithLabel.getCursor();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getDraggable", function () {
      return _this.state.markerWithLabel.getDraggable();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getIcon", function () {
      return _this.state.markerWithLabel.getIcon();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getLabel", function () {
      return _this.state.markerWithLabel.getLabel();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getOpacity", function () {
      return _this.state.markerWithLabel.getOpacity();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getPlace", function () {
      return _this.state.markerWithLabel.getPlace();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getPosition", function () {
      return _this.state.markerWithLabel.getPosition();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getShape", function () {
      return _this.state.markerWithLabel.getShape();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getTitle", function () {
      return _this.state.markerWithLabel.getTitle();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getVisible", function () {
      return _this.state.markerWithLabel.getVisible();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getZIndex", function () {
      return _this.state.markerWithLabel.getZIndex();
    });

    return _this;
  }

  return MarkerWithLabel;
}(_react.PureComponent);

exports.MarkerWithLabel = MarkerWithLabel;

_defineProperty(MarkerWithLabel, "contextType", _MarkerCluster.MarkerClusterContext);

var _default = MarkerWithLabel;
exports.default = _default;
