// react-google-maps-api/lib/GoogleMap.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.GoogleMap = void 0;

var _react = _interopRequireWildcard(require("react"));

var _proptypes = require("./proptypes");

var _mapContext = _interopRequireDefault(require("./map-context"));

var _instancePersistance = require("./utils/instance-persistance");

var _helper = require("./utils/helper");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }

var eventMap = {
  onDblClick: 'dblclick',
  onDragEnd: 'dragend',
  onDragStart: 'dragstart',
  onMapTypeIdChanged: 'maptypeid_changed',
  onMouseMove: 'mousemove',
  onMouseOut: 'mouseout',
  onMouseOver: 'mouseover',
  onRightClick: 'rightclick',
  onTilesLoaded: 'tilesloaded',
  onBoundsChanged: 'bounds_changed',
  onCenterChanged: 'center_changed',
  onClick: 'click',
  onDrag: 'drag',
  onHeadingChanged: 'heading_changed',
  onIdle: 'idle',
  onProjectionChanged: 'projection_changed',
  onResize: 'resize',
  onTiltChanged: 'tilt_changed',
  onZoomChanged: 'zoom_changed'
};
var updaterMap = {
  extraMapTypes: function extraMapTypes(map, extra) {
    extra.forEach(function (it) {
      var _map$mapTypes;

      (_map$mapTypes = map.mapTypes).set.apply(_map$mapTypes, _toConsumableArray(it));
    });
  },
  center: function center(map, _center) {
    map.setCenter(_center);
  },
  clickableIcons: function clickableIcons(map, icons) {
    map.setClickableIcons.apply(map, _toConsumableArray(icons));
  },
  heading: function heading(map, _heading) {
    map.setHeading(_heading);
  },
  mapTypeId: function mapTypeId(map, _mapTypeId) {
    map.setMapTypeId(_mapTypeId);
  },
  options: function options(map, _options) {
    map.setOptions(_options);
  },
  streetView: function streetView(map, _streetView) {
    map.setStreetView(_streetView);
  },
  tilt: function tilt(map, _tilt) {
    map.setTilt(_tilt);
  },
  zoom: function zoom(map, _zoom) {
    map.setZoom(_zoom);
  }
};

var GoogleMap =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(GoogleMap, _PureComponent);

  function GoogleMap() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, GoogleMap);

    for (var _len = arguments.length, _args = new Array(_len), _key = 0; _key < _len; _key++) {
      _args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(GoogleMap)).call.apply(_getPrototypeOf2, [this].concat(_args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "state", {
      map: null
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "registeredEvents", []);

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getInstance", function () {
      var _this$props = _this.props,
          reuseSameInstance = _this$props.reuseSameInstance,
          rest = _objectWithoutProperties(_this$props, ["reuseSameInstance"]);

      var map = reuseSameInstance && (0, _instancePersistance.restoreInstance)(rest);
      return map || new google.maps.Map(_this.mapRef, _objectSpread({}, _this.props.options));
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidMount", function () {
      _this.setState(function () {
        return {
          map: _this.getInstance()
        };
      }, function () {
        _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
          updaterMap: updaterMap,
          eventMap: eventMap,
          prevProps: {},
          nextProps: _this.props,
          instance: _this.state.map
        });

        _this.props.onLoad(_this.state.map);
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentDidUpdate", function (prevProps) {
      (0, _helper.unregisterEvents)(_this.registeredEvents);
      _this.registeredEvents = (0, _helper.applyUpdatersToPropsAndRegisterEvents)({
        updaterMap: updaterMap,
        eventMap: eventMap,
        prevProps: prevProps,
        nextProps: _this.props,
        instance: _this.state.map
      });
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "componentWillUnmount", function () {
      var _this$props2 = _this.props,
          reuseSameInstance = _this$props2.reuseSameInstance,
          id = _this$props2.id;
      reuseSameInstance && (0, _instancePersistance.saveInstance)(id, _this.state.map);
      (0, _helper.unregisterEvents)(_this.registeredEvents);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getRef", function (ref) {
      _this.mapRef = ref;
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "render", function () {
      var _this$props3 = _this.props,
          id = _this$props3.id,
          mapContainerStyle = _this$props3.mapContainerStyle,
          mapContainerClassName = _this$props3.mapContainerClassName,
          children = _this$props3.children;
      var map = _this.state.map;
      return _react.default.createElement("div", {
        id: id,
        ref: _this.getRef,
        style: mapContainerStyle,
        className: mapContainerClassName
      }, _react.default.createElement(_mapContext.default.Provider, {
        value: map
      }, map !== null ? children : null));
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "fitBounds", function () {
      var _this$state$map;

      return (_this$state$map = _this.state.map).fitBounds.apply(_this$state$map, arguments);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "panBy", function () {
      var _this$state$map2;

      return (_this$state$map2 = _this.state.map).panBy.apply(_this$state$map2, arguments);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "panTo", function () {
      var _this$state$map3;

      return (_this$state$map3 = _this.state.map).panTo.apply(_this$state$map3, arguments);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "panToBounds", function () {
      var _this$state$map4;

      return (_this$state$map4 = _this.state.map).panToBounds.apply(_this$state$map4, arguments);
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getBounds", function () {
      return _this.state.map.getBounds();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getCenter", function () {
      return _this.state.map.getCenter();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getClickableIcons", function () {
      return _this.state.map.getClickableIcons();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getDiv", function () {
      return _this.state.map.getDiv();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getHeading", function () {
      return _this.state.map.getHeading();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getMapTypeId", function () {
      return _this.state.map.getMapTypeId();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getProjection", function () {
      return _this.state.map.getProjection();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getStreetView", function () {
      return _this.state.map.getStreetView();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getTilt", function () {
      return _this.state.map.getTilt();
    });

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getZoom", function () {
      return _this.state.map.getZoom();
    });

    return _this;
  }

  return GoogleMap;
}(_react.PureComponent);

exports.GoogleMap = GoogleMap;

_defineProperty(GoogleMap, "defaultProps", {
  id: 'defaultMapId',
  reuseSameInstance: false,
  onLoad: function onLoad() {}
});

var _default = GoogleMap;
exports.default = _default;
